### DISCUSSION TITLE
[2장] 웹사이트 접근 요청을 받고 브라우저가 화면을 그리기까지의 과정을 설명해주세요.

### DISCUSSION THREAD

**Discussion by codefug** [FE-tech-talk/TechTalk-React#11] [Q&A] [answered]
ㅈㄱㄴ

**Comment by easyhyun00** [comment, marked as answer]
## 답변

사용자가 브라우저에 들어옴
서버는 그 브라우저의 url로 http 요청을 함
서버가 html 파일을 반환
브라우저가 반환된 html 파싱
DOM 생성
css 만나면 해당 css 다운하고 CSSOM 생성
눈에 보이는 노드만 렌더링
해당 노드에 대한 CSSOM 적용

**Comment by codefug** [comment]
1. 사용자가 주소를 입력해서 서버로부터 HTML을 다운 받는다.
2. 브라우저가 렌더링 엔진에서 HTML을 DOM트리로 파싱한다. 이 과정에서 link 같은 tag에서 발견되는 css파일은 다운로드한다.
3. 브라우저가 렌더링 엔진에서 CSS를 CSSOM트리로 파싱한다.
4. DOM트리를 순회한다 CSSOM에 정의되어 있는 보이지 않는 요소들은 순회하지 않는다.
5. 보이지 않는 요소들을 제외하고 CSS를 입힌다. 순서는 layout > paint 순서이다.
